compiler/src/dotty/tools/dotc/CompilationUnit.scala

```diff
@@ -38,15 +38,26 @@ class CompilationUnit protected (val source: SourceFile) {
    */
   val freshNames: FreshNameCreator = new FreshNameCreator.Default
 
+  /** Will be set to `true` if there are inline call that must be inlined after typer.
+   *  The information is used in phase `Inlining` in order to avoid traversing trees that need no transformations.
+   */
+  var needsInlining: Boolean = false
+
   /** Will be set to `true` if contains `Quote`.
    *  The information is used in phase `Staging` in order to avoid traversing trees that need no transformations.
    */
   var needsStaging: Boolean = false
 
+  /** Will be set to `true` if contains `Quote` that needs to be pickled
+   *  The information is used in phase `PickleQuotes` in order to avoid traversing trees that need no transformations.
+   */
+  var needsQuotePickling: Boolean = false
+
```

Comment:
Do we need both `needsStaging` and `needsQuotePickling`?

Reply:
There are some cases where we only need one of them. Each causes a full tree traversal as these are `MacroTransform` phases.